static/app/utils/profiling/fzf/fzf.ts

```diff
@@ -0,0 +1,233 @@
+// Port of the fzf v1 algorithm
+// https://github.com/junegunn/fzf/blob/f81feb1e69e5cb75797d50817752ddfe4933cd68/src/algo/algo.go#L615
+
+interface Result {
+  readonly end: number;
+  readonly matches: ReadonlyArray<[number, number]>;
+  readonly score: number;
+  readonly start: number;
+}
+
+// https://github.com/junegunn/fzf/blob/f81feb1e69e5cb75797d50817752ddfe4933cd68/src/algo/algo.go#L107
+const scoreMatch = 16;
+const scoreGapStart = -3;
+const scoreGapExtention = -1;
+const bonusBoundary = scoreMatch / 2;
+const bonusNonWord = scoreMatch / 2;
+const bonusCamel123 = bonusBoundary + scoreGapExtention;
+const bonusConsecutive = -(scoreGapStart + scoreGapExtention);
+const bonusFirstCharMultiplier = 2;
+
+enum CharTypes {
+  charLower,
+  charUpper,
+  charNumber,
+  charNonWord,
+}
+
+const CharCodes = {
+  lowerA: 'a'.charCodeAt(0),
+  lowerZ: 'z'.charCodeAt(0),
+  upperA: 'A'.charCodeAt(0),
+  upperZ: 'Z'.charCodeAt(0),
+  zero: '0'.charCodeAt(0),
+  nine: '9'.charCodeAt(0),
+};
+
+function getCharClass(c: number): CharTypes {
+  if (c >= CharCodes.lowerA && c <= CharCodes.lowerZ) {
+    return CharTypes.charLower;
+  }
+  if (c >= CharCodes.upperA && c <= CharCodes.upperZ) {
+    return CharTypes.charUpper;
+  }
+  if (c >= CharCodes.zero && c <= CharCodes.nine) {
+    return CharTypes.charNumber;
+  }
+  return CharTypes.charNonWord;
+}
+
+// Algo functions make two assumptions
+// 1. "pattern" is given in lowercase if "caseSensitive" is false
+// 2. "pattern" is already normalized if "normalize" is true
+// https://github.com/junegunn/fzf/blob/f81feb1e69e5cb75797d50817752ddfe4933cd68/src/algo/algo.go#L244
+export function fzf(text: string, pattern: string, caseSensitive: boolean): Result {
+  if (pattern.length === 0) {
+    return {end: 0, score: 0, start: 0, matches: []};
+  }
+
+  let pidx = 0;
+  let sidx = -1;
+  let eidx = -1;
+
+  const textLength = text.length;
+  const patternLength = pattern.length;
+
+  for (let index = 0; index < textLength; index++) {
+    let char = text[index];
+    // This is considerably faster than blindly applying strings.ToLower to the whole string
+    if (!caseSensitive) {
+      const cc = char.charCodeAt(0);
+      if (cc >= 65 && cc <= 90) {
```

Comment:
what are these char codes?

Reply:
A and Z -> we convert to uppercase to lowercase here